Build failure with Clang4

Build fails on FreeBSD with Clang 4.0 with following error:
```
bushnell.cc:138:36: error: ordered comparison between pointer and zero ('const char *' and 'int')
  for (t = bushnell_icons; t->icon > 0; t++) {
                           ~~~~~~~ ^ ~
bushnell.cc:150:36: error: ordered comparison between pointer and zero ('const char *' and 'int')
  for (t = bushnell_icons; t->icon > 0; t++) {
                           ~~~~~~~ ^ ~
```

Full build log [here](http://sprunge.us/MYYK)

I'll look to patch the FreeBSD port to resolve this. I created this issue to ensure upstream knows about it.

--- a/bushnell.cc
+++ b/bushnell.cc
@@ -136,7 +136,7 @@ bushnell_get_icon_from_name(QString name)
     name = "Waypoint";
   }
 
-  for (t = bushnell_icons; t->icon > 0; t++) {
+  for (t = bushnell_icons; t->icon != nullptr; t++) {
     if (0 == name.compare(t->icon, Qt::CaseInsensitive)) {
       return t->symbol;
     }
@@ -148,7 +148,7 @@ static const char*
 bushnell_get_name_from_symbol(signed int s)
 {
   icon_mapping_t* t;
-  for (t = bushnell_icons; t->icon > 0; t++) {
+  for (t = bushnell_icons; t->icon != nullptr; t++) {
     if (s == t->symbol) {
       return t->icon;
     }

